vuu-ui/packages/vuu-shell/test/layout-management/useLayoutManager.test.tsx

```diff
@@ -0,0 +1,209 @@
+    ...actual,
+    isLayoutJSON: vi.fn(),
+    resolveJSONPath: vi.fn(),
+  };
+});
+
+const wrapper = ({ children }) => (
+  <LayoutManagementProvider>{children}</LayoutManagementProvider>
+);
+
+describe("LayoutManagementProvider", () => {
+  let persistence: PersistenceManager;
+
+  beforeEach(() => {
+    persistence = new LocalPersistenceManager();
+    vi.mocked(persistence.loadMetadata).mockResolvedValueOnce([]);
+    vi.mocked(persistence.loadApplicationJSON).mockResolvedValueOnce(
+      initialApplicationJSON
+    );
+    vi.spyOn(global.console, "error");
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  it("calls loadMetadata and loadApplicationJSON on mount", () => {
+    const { result } = renderHook(() => useLayoutManager(), { wrapper });
+
+    expect(persistence.loadMetadata).toHaveBeenCalled();
+    expect(persistence.loadApplicationJSON).toHaveBeenCalled();
+    expect(result.current.applicationJson).toBe(initialApplicationJSON);
+  });
+
+  describe("saveApplicationLayout", () => {
+    it("calls saveApplicationJSON when layout is valid", () => {
+      const { result } = renderHook(useLayoutManager, { wrapper });
+
+      vi.mocked(persistence.saveApplicationJSON).mockResolvedValueOnce();
+      vi.mocked(isLayoutJSON).mockReturnValue(true);
+
+      result.current.saveApplicationLayout(newLayout);
+
+      expect(persistence.saveApplicationJSON).toHaveBeenCalledWith({
+        ...initialApplicationJSON,
+        layout: newLayout,
+      });
+    });
+
+    it("doesn't call saveApplicationJSON and logs error when layout is not valid ", () => {
+      const { result } = renderHook(() => useLayoutManager(), { wrapper });
+
+      vi.mocked(persistence.saveApplicationJSON).mockResolvedValueOnce();
+      vi.mocked(isLayoutJSON).mockReturnValue(false);
+
+      result.current.saveApplicationLayout(newLayout);
+
+      expect(persistence.saveApplicationJSON).not.toHaveBeenCalled();
+      expect(console.error).toHaveBeenCalledWith(
+        "Tried to save invalid application layout",
+        newLayout
+      );
+    });
+  });
+
+  describe("saveLayout", () => {
+    it("calls createLayout when layout is valid and path is resolved", () => {
+      const { result } = renderHook(() => useLayoutManager(), { wrapper });
+
+      vi.mocked(persistence.createLayout).mockResolvedValueOnce(metadata);
+      vi.mocked(isLayoutJSON).mockReturnValue(true);
+      vi.mocked(resolveJSONPath).mockReturnValue(newLayout);
+
+      result.current.saveLayout(metadata);
+
+      expect(persistence.createLayout).toHaveBeenCalled();
+    });
+
+    it("doesn't call createLayout, triggers error notification and logs error when layout path can't be resolved ", () => {
+      const { result } = renderHook(() => useLayoutManager(), { wrapper });
+      const { notify } = useNotifications();
+
+      vi.mocked(persistence.createLayout).mockResolvedValueOnce(metadata);
+      vi.mocked(resolveJSONPath).mockReturnValue(undefined);
```

Comment:
Might be good to mock a positive result for `isLayoutJSON` here, just to ensure that has no bearing on the outcome of this test.

Reply:
good point

Reply:
done